### PR TITLE
Added notes re mahal and changed its IQR func

### PR DESCRIPTION
**Pull request by aac61**
@matthew-brett - our analysis_plan.md is ready for review.

### CODE REVIEW

File: analysis_plan.md [hunk 1]
```diff
@@ -15,18 +15,17 @@
...
-This metric does not have a clear/intuitive interpretation; it flags out a volume as "different", but we do not know why or how that volume differs.  
+The basic DVARS function seems limited because of all the issues discussed in the "ontheproject.md" file. We tried a generalized version of DVARS (dvars_all in metrics.py called via find_outliers), which compares each volumes against all of the others (not just the exact previous one) and generates a metric of how different a given volume is from the "center" of the volumes (the values can be tentatively found in a file called "potential_outliers.txt" in the main folder). This metric is computationally very complex and the computation takes a long time to run (approx 30 minutes!). We need to come up with a more efficient way? This metric does not have a clear/intuitive interpretation; it flags out a volume as "different", but we do not know why or how that volume differs.
```
Comment (matthew-brett) [inline analysis_plan.md:18]: I guess you could try something simpler.  For example, you take the mean volume and compare against that.  There are various problems with that, of which two obvious ones are that the mean includes outliers, and the person may move slowly during the scan session, so first and last volumes will look particularly different.

File: analysis_plan.md [hunk 2]
```diff
@@ -15,18 +15,17 @@
...
 
 An alternative is to use the simpler SPM function, which is much more easily interpretable and produces a set of scans that have already some of the empty space around the head removed (important because this space could introduce noise). 
 
-Andromachi has added some a function called mahal() that uses Mahalanobis distances to account for correlation. The results are _very_ similar to the generalised dvars function and runs way faster! 
+Andromachi has added some a function called mahal() that uses Mahalanobis distances to account for correlation. The results are _very_ similar to the generalised dvars function and runs way faster (about 2 min)! As an added bonus it removes the same set of 'non-brain' voxels from every scan before computing its index. As the generalised dvars function though, the end value is an index of diviation from the 'center' (correct me if I am wrong Andromachi). So we cannot imidiatly use this metrix to understand why a particular scan is an outlier. That would require some inspection. Again similar to the generalised dvars function, we should not use the IQR to remove values below the first quantile (that would remove our most centered data!).
```
Comment (matthew-brett) [inline analysis_plan.md:22]: Yes, you need a unidirectional threshold here.

File: findoutlie/detectors.py [hunk 3]
```diff
@@ -61,7 +61,7 @@ def iqr_detector(measures, iqr_proportion=1.5):
     return outlier_tf
     #we only want over if we use dvars
 
-def iqr_detector_for_dvars_all(measures, iqr_proportion=1.5):
+def iqr_detector_uper_only(measures, iqr_proportion=1.5):
```
Comment (matthew-brett) [inline findoutlie/detectors.py:64]:
```suggestion
def iqr_detector_upper_only(measures, iqr_proportion=1.5):
```

**Review by matthew-brett — COMMENTED** [review]
Thanks - a good start.   But consider the problem of slow drift - which is typical in FMRI scans.  This means that you are looking for sudden shifts in signal compared to the scans that are nearby in time.

**Comment by matthew-brett** [comment]
@effigies @oesteban - any thoughts here?

File: analysis_plan.md (hunk 1 above)
Comment (oesteban) [reply analysis_plan.md:18]: For the problem of bias on the mean estimator you can use median instead, which is unbiased and gives you the same reference you need more robustly.

Comment (oesteban) [reply analysis_plan.md:18]: However, if you don't want to deviate much from DVARS, Tom Nichols proposed a variant of DVARS, called standardized DVARS, which can be used to more easily set a more reliable threshold (i.e., that works on new data)

**Comment by effigies** [comment]
I think everything Matthew and Oscar have said is sensible, so I'll try not to cover the same ground. I'm looking at the [overall analysis plan](https://github.com/aac61/diagnostics-let_the_python_eat_us/blob/main/analysis_plan.md) that includes:

> We have looked into a number of papers discussing outlier detection in neuroimaging scans. However, we are at this point lacking the theoretical & mathematical knowledge to implement the formulas and were unable to find relevant code/examples. Any help would be greatly appreciated. @nipraxis-fall-2022/instructors
> 
> The materials we have looked into thus far are the following:
> 
> https://www-sciencedirect-com.ezp.lib.cam.ac.uk/science/article/pii/S1361841512000564 https://www.sciencedirect.com/science/article/pii/S1053811917311229?via%3Dihub https://canlab.github.io/_pages/tutorials/html/nuisance_covariates.html https://wiki.cam.ac.uk/bmuwiki/Useful_Code https://towardsdatascience.com/exploring-cognitive-differences-via-resting-state-networks-2112bf5291e2

Some notes:

1) For the Fritsch et al regularized MCD, I probably would skip over this one for now. It would probably be instructive to work through, but I would work on algorithms that have algebraic expressions before moving on to optimization methods.
2) For Afyouni and Nichols, you've already calculated DVARS, and alternative formulations are listed in table 3. I would probably ignore the rest of the paper except for the analyses of which work better in which situations.
3) The CANLAB docs work off of the fMRIPrep confounds outputs. You have the means of calculating `(std_)dvars`.
   * If you want to calculate RMSD or FD, you'll need to run motion correction and save the rotation and translation parameters. Once you have those, though, the formulae are quite straightforward. As a suggestion, running MCFLIRT once and saving the motion parameters in the repository will help avoid the expensive computations.
   * If you want to calculate non-steady-state outliers, you'll need an algorithm that takes into account [the phenomenon](https://mriquestions.com/how-is-signal-higher.html) you're trying to capture. Not in the sense of modeling the exponential decay but in the coarser sense of which volumes could potentially be affected.
4) Finally the last article looks at MRIQC outputs. You can find formulae for most of these: https://mriqc.readthedocs.io/en/latest/iqms/bold.html

If there are notational questions for how to interpret a particular formula (many use notation that obscures the summations), I'm happy to answer them. As a rule, we should be able to recast most expressions into terms and operators that correspond to `np.diff()` `np.sum()`, `np.mean()` and `np.std()` and `np.linalg.pinv()`.

**Pull request merged**